gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/publisher/HiveConvertPublisher.java

```diff
@@ -57,13 +67,43 @@ public void initialize() throws IOException {
   @Override
   public void publishData(Collection<? extends WorkUnitState> states) throws IOException {
 
+    String cleanupCommands = StringUtils.EMPTY;
+    boolean isFirst = true;
     if (Iterables.tryFind(states, UNSUCCESSFUL_WORKUNIT).isPresent()) {
       for (WorkUnitState wus : states) {
+        if (isFirst) {
+          // Get cleanup staging table commands
+          cleanupCommands = HiveAvroORCQueryUtils.GSON
+              .fromJson(wus.getProp(HiveAvroORCQueryUtils.SERIALIZED_CLEANUP_COMMANDS), String.class);
+          isFirst = false;
+        }
         wus.setWorkingState(WorkingState.FAILED);
         new SlaEventSubmitter(eventSubmitter, EventConstants.CONVERSION_FAILED_EVENT, wus.getProperties()).submit();
       }
     } else {
+      String publishTableCommands = StringUtils.EMPTY;
       for (WorkUnitState wus : states) {
+        if (isFirst) {
+          // Get publish table commands
+          publishTableCommands = HiveAvroORCQueryUtils.GSON
```

Comment:
I think reading and writing the publish commands should be hidden inside ``` HiveAvroORCQueryUtils```. The publisher and converter should not know the workunit key or the serialization semantics.